.github/workflows/ironic-image-build-pr.yml

```diff
@@ -0,0 +1,43 @@
+name: Build Ironic Image on PRs
+
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize, ready_for_review]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        centos_version:
+          - stream9
+          - stream10
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3.11.1
+
+      - name: Set base image
+        id: base-image
+        run: |
+          if [[ "${{ matrix.centos_version }}" == "stream9" ]]; then
+            echo "base_image=quay.io/centos/centos:stream9" >> "${GITHUB_OUTPUT}"
+          elif [[ "${{ matrix.centos_version }}" == "stream10" ]]; then
+            echo "base_image=quay.io/centos/centos:stream10" >> "${GITHUB_OUTPUT}"
+          fi
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v6.18.0
```

Comment:
Must be pinned with a sha. See all the other workflow files.

Reply:
alright